integrations/gitlab/gitlab_integration/git_integration.py

```diff
@@ -1,47 +1,125 @@
+import typing
 from functools import lru_cache
 from typing import Dict, Any, Tuple, List
 
-from gitlab_integration.core.entities import FILE_PROPERTY_PREFIX
-from gitlab_integration.gitlab_service import GitlabService
+from gitlab_integration.core.entities import (
+    FILE_PROPERTY_PREFIX,
+    SEARCH_PROPERTY_PREFIX,
+)
+from gitlab_integration.gitlab_service import PROJECTS_CACHE_KEY
 from loguru import logger
 from pydantic import Field
+from gitlab.v4.objects import Project
 
 from port_ocean.context.event import event
 from port_ocean.core.handlers import JQEntityProcessor
 from port_ocean.core.handlers.port_app_config.models import PortAppConfig
 
 
-class GitManipulationHandler(JQEntityProcessor):
+class FileEntityProcessor(JQEntityProcessor):
```

Comment:
Im not sure there is a need for a full class for this
They can be simple functions